src/main/java/frc/robot/subsystems/SuperStructure.java

```diff
@@ -168,6 +181,85 @@ public Command coralLevelOne(BooleanSupplier score) {
         .withName("Coral Level 1");
   }
 
+  // New versions with ipScore
+  // if robot is in position for intermediate scoring, it can score early but if vision is dead
+  // manual control still works
+  // only used on solo controls
+
+  public Command coralLevelThree(BooleanSupplier score, BooleanSupplier ipScore) { // same as L4
+    return Commands.sequence(
+            Commands.parallel(
+                    elevator
+                        .setLevel(ElevatorSubsystem.CORAL_LEVEL_THREE_PRE_POS)
+                        .deadlineFor(
+                            armPivot
+                                .moveToPosition(ArmPivot.CORAL_PRESET_UP)
+                                .until(ipScore)
+                                .until(score)),
+                    spinnyClaw.stop())
+                .withTimeout(0.5),
+            repeatPrescoreScoreSwing(
+                Commands.repeatingSequence(
```

Comment:
Seems odd to have a repeating sequence passed in to a function that repeats the passed in command. Is that intentional?

Reply:
Other than that, this is identical to `coralLevelTwo()` but with different constants for a few things. If they should match other than constants, it'd be useful to rework this post-comp to remove the duplicated code (via a template function or some other method)